src/hal/drivers/mesa-hostmot2/bspi.c

```diff
@@ -197,9 +197,21 @@ int hm2_bspi_write_chan(char* name, int chan, rtapi_u32 val)
     return r;
 }
 
+int hm2_bspi_clear_counter(hostmot2_t *hm2, int i)
+{
+    rtapi_u32 buff = 0; // data is dont care when writing to count register
+    int r;
+    r = hm2->llio->write(hm2->llio, hm2->bspi.instance[i].count_addr, &buff, sizeof(rtapi_u32));
+    if (r < 0) {
+        HM2_ERR("BSPI: hm2->llio->write failure %s\n", hm2->bspi.instance[i].name);
+    }
+
+    return r;
+}
+
 EXPORT_SYMBOL_GPL(hm2_bspi_setup_chan);
 int hm2_bspi_setup_chan(char *name, int chan, int cs, int bits, float mhz,
-                        int delay, int cpol, int cpha, int clear, int echo)
+                        int delay, int cpol, int cpha, int clear, int echo, int samplelate)
```

Comment:
you change the function signature here, but don't change the declaration or use sites until a later commit.  Please move to the appropriate commit.

Symptom: Compiling at this ref,
~~~~
hal/drivers/mesa-hostmot2/bspi.c:213:5: error: conflicting types for ‘hm2_bspi_setup_chan’
 int hm2_bspi_setup_chan(char *name, int chan, int cs, int bits, float mhz,
     ^~~~~~~~~~~~~~~~~~~
In file included from hal/drivers/mesa-hostmot2/bspi.c:25:0:
hal/drivers/mesa-hostmot2/hostmot2.h:1304:5: note: previous declaration of ‘hm2_bspi_setup_chan’ was here
 int hm2_bspi_setup_chan(char *name, int chan, int cs, int bits, float mhz,
     ^~~~~~~~~~~~~~~~~~~
Makefile:1005: recipe for target 'objects/rthal/drivers/mesa-hostmot2/bspi.o' failed
~~~~